### PR TITLE
Declare tasks programatically rather than declaratively

### PR DESCRIPTION
**Pull request by nex3**
This has a few benefits:

* It allows us to dynamically choose task dependencies (so for example
  the Linux standalone package can depend on the native executable
  task when running on Linux).

* Similar tasks can be defined programmatically.

* Task names are decoupled from Dart function names, which in turn
  allows us to avoid manually namespacing the package in favor of
  encouraging it to be imported with a Dart namespace.

* It works around google/grinder.dart#337 and google/grinder.dart#338.

### CODE REVIEW

File: lib/src/standalone.dart [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+/// Returns whether to use the natively-compiled executable for the given [os]
+/// and architecture combination.
+///
+/// We can only use the native executable on the current operating system *and*
+/// on 64-bit machines, because currently Dart doesn't support cross-compilation
+/// (dart-lang/sdk#28617) and only 64-bit Dart SDKs ship with `dart2aot`.
+bool _useNative(String os, {@required bool x64}) =>
+    os == Platform.operatingSystem && x64 == _is64Bit;
+
+/// Builds a Sass package for the given [os] and architecture.
```
Comment (srawlins) [inline lib/src/standalone.dart]: Remove the word Sass?

File: lib/src/standalone.dart [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+
+  var filename = "/bin/dart${os == 'windows' ? '.exe' : ''}";
+  return ZipDecoder()
+      .decodeBytes(response.bodyBytes)
+      .firstWhere((file) => file.name.endsWith(filename))
+      .content as List<int>;
+}
+
+/// Returns the architecture name for the given boolean.
+String _arch(bool x64) => x64 ? "x64" : "ia32";
```
Comment (srawlins) [inline lib/src/standalone.dart]: binary extension might be another good helper here. `os == 'windows' ? '.exe' : ''` appears three times in this file, and all three are in string interpolations. :/

File: lib/src/standalone.dart (hunk 1 above)
Comment (nex3) [reply lib/src/standalone.dart]: Done.

File: lib/src/standalone.dart (hunk 2 above)
Comment (nex3) [reply lib/src/standalone.dart]: Done.

**Pull request merged**